packages/grpc-js/src/server-call.ts

```diff
@@ -255,6 +258,15 @@ export class ServerDuplexStreamImpl<RequestType, ResponseType> extends Duplex
   sendMetadata(responseMetadata: Metadata): void {
     this.call.sendMetadata(responseMetadata);
   }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  end(metadata?: any) {
```

Comment:
I know this is a copy from below but shouldn't this be `metadata?: Metadata` ?
`this.trailingMetadata` is of type Metadata so I can't imagine what other type it could be.

Reply:
The problem is that TypeScript doesn't like having a class that extends `Duplex` but implements `end` with a type that is incompatible with the `end` method declared in the `Duplex` class. Declaring the argument as `any` smooths that out.

There's a more complicated way of doing this that I use in [object-stream.ts](https://github.com/grpc/grpc-node/blob/master/packages/grpc-js/src/object-stream.ts) where I declare an intermediate type that uses `any & T` as the type, and then I can subtype that using `T`. I don't think that's valuable here though because this type is only in the implementation. The interface that actually gets provided to the user has the stricter `Metadata` type, and that's what matters here.

Reply:
@murgatroid99 The other day I came across a similar situation and was able to mitigate it with `Omit<Type, 'end'>`. Might be useful in the future.